File: src/main/java/com/example/chatService/oauth/CustomOAuth2User.java
package com.example.chatService.oauth;

import com.example.chatService.entity.User;
import lombok.AllArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.oauth2.core.user.OAuth2User;

import java.util.Collection;
import java.util.Map;

@AllArgsConstructor
public class CustomOAuth2User implements OAuth2User {

    private final User user;
    private final Map<String, Object> attributes;

    public User getUser() {
        return user;
    }

    @Override
    public Map<String, Object> getAttributes() {
        return attributes;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return user.getAuthorities(); // ROLE_USER
    }

    @Override
    public String getName() {
        return user.getId(); // OAuth2User 식별자
    }
}

File: src/main/java/com/example/chatService/oauth/CustomOAuth2User.kt
package com.example.chatService.oauth

import com.example.chatService.entity.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.oauth2.core.user.OAuth2User

class CustomOAuth2User(
    private val user: User,
    private val attributes: Map<String, Any>,
) : OAuth2User {
    fun getUser(): User = user

    override fun getAttributes(): Map<String, Any> = attributes

    override fun getAuthorities(): Collection<GrantedAuthority> = user.authorities

    override fun getName(): String = user.id ?: ""
}
